src/main/java/org/apache/aurora/scheduler/configuration/ConfigurationManager.java

```diff
@@ -170,12 +164,24 @@ public IJobConfiguration validateAndPopulate(IJobConfiguration job)
       throw new TaskDescriptionException("Job configuration must have taskConfig set.");
     }
 
+    JobConfiguration builder = job.newBuilder();
+    
     if (job.getInstanceCount() <= 0) {
       throw new TaskDescriptionException("Instance count must be positive.");
+    } else {
+      ImmutableList<IInstance> instances = job.getTaskConfig().getInstances();
+      if (!instances.isEmpty()) {
+        if (job.getInstanceCount() == 1) { // replace the default instanceCount
```

Comment:
good point, the default 1 is set at the python client level. I'd change the default there to 0 and use that in this condition. If we find 0 here (and no instances), we just set instanceCount to 1. Makes sense?